src/OpenQasmReader/Parser.cs

```diff
@@ -326,7 +326,7 @@ private static void ParseGateCall(IEnumerator<string> token, StringBuilder build
             {
                 Indent(builder);
                 var size = qubits.First(q => !q.Contains('['));
-                builder.AppendFormat("for (_idx in 0 .. Length({0})) {{\n", size);
+                builder.AppendFormat("for (_idx in 0 .. Length({0}) - 1) {{\n", size);
```

Comment:
Ah yes... I've made that mistake so many times. We should change that to other types of applies though.